libpromises/locks.c

```diff
@@ -365,27 +365,91 @@ static int RemoveLock(const char *name)
     return 0;
 }
 
+static bool NoOrObsoleteLock(LockData *entry, ARG_UNUSED size_t val_size, size_t *max_old)
```

Comment:
`val_size` is not unused, and should be called `entry_size` IMO.

Reply:
> `val_size` is not unused, and should be called `entry_size` IMO.

It's unused in non-debug builds.